cudax/include/cuda/experimental/__cuco/detail/__config

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2022-2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef _CUDAX_DETAIL_CONFIG
```

Comment:
This header guard should be `_CUDAX___CUCO_CONFIG`

We do not use detail folders inside the already implementation defined folder `__cuco` so please move everything you need into `__cuco` or `__cuco/has_functions`